### PR TITLE
Ansible Catchup

### PR DESCRIPTION
**Pull request by rambleraptor**
The final catchup PR! This merges in Ansible commits, brings in Ansible resource override support, and fixes some misc bugs on Puppet/Chef DNS.

Corresponding Ansible PR: https://github.com/modular-magician/ansible/pull/1/commits

Take a look at the last two commits in particular, since that's new code. This is a large PR, so let me know how to best split this up.

-----------------------------------------------------------------
# [all]
Ansible Catchup as of 20180501
### [puppet-dns]
Moving quota to NestedObject

### CODE REVIEW

**Comment by ndmckinley** [comment]
This PR crashes terraform generation (so the magician will never show up to comment).  Any idea what might be going on there?

**Comment by rambleraptor** [comment]
Good call. Not sure how I missed that last time around.

Made a change that I'm fairly positive fixes it. Made a poor assumption about how overrides are working.

File: templates/ansible/resource.erb [hunk 1]
```diff
@@ -272,11 +269,11 @@ def resource_to_request(module):
 <%
   encoder_name = if object.encoder?
                    object.transport.encoder
-                 else config['encoder']
-                   config['encoder']
+                 else object.encoder
```
Comment (rosbo) [inline templates/ansible/resource.erb]: Why is there a condition for a else statement here?

File: .gitmodules [hunk 2]
```diff
@@ -66,7 +66,8 @@
 	url = git@github.com:GoogleCloudPlatform/chef-google-logging
 [submodule "build/ansible"]
 	path = build/ansible
-	url = git@github.com:ansible/ansible.git
+	url = git@github.com:modular-magician/ansible
```
Comment (nelsonjr) [inline .gitmodules]: fix tabs/spaces.

File: products/dns/api.yaml [hunk 3]
```diff
@@ -89,39 +89,44 @@ objects:
...
-        description: |
-          Maximum allowed number of ResourceRecordSets per zone in the
-          project.
-        output: true
-      - !ruby/object:Api::Type::Integer
-        name: 'quota.totalRrdataSizePerChange'
-        description: |
-          Maximum allowed size for total rrdata in one ChangesCreateRequest
-          in bytes.
+      - !ruby/object:Api::Type::NestedObject
```
Comment (nelsonjr) [inline products/dns/api.yaml]: You need to add an entry to `puppet.yaml` and `chef.yaml` to warn about this change.

File: provider/ansible/resource_override.rb [hunk 4]
```diff
@@ -28,8 +42,18 @@ def validate
...
 
         default_value_property :access_api_results, false
+        default_value_property :aliases, {}
+        default_value_property :decoder, false
+        default_value_property :exclude, false
+        default_value_property :editable, true
+        default_value_property :imports, []
+        default_value_property :provider_helpers, []
 
         check_property :access_api_results, :boolean
```
Comment (nelsonjr) [inline provider/ansible/resource_override.rb]: Check all properties.

File: products/dns/api.yaml (hunk 3 above)
Comment (rambleraptor) [reply products/dns/api.yaml]: What did you mean by this?

File: .gitmodules (hunk 2 above)
Comment (rambleraptor) [reply .gitmodules]: Fixed.

File: provider/ansible/resource_override.rb (hunk 4 above)
Comment (rambleraptor) [reply provider/ansible/resource_override.rb]: Fixed.

File: products/compute/ansible.yaml [hunk 5]
```diff
@@ -25,19 +25,43 @@ manifest: !ruby/object:Provider::Ansible::Manifest
   author: Google Inc. (@googlecloudplatform)
 # This is where custom code would be defined eventually.
 overrides: !ruby/object:Provider::ResourceOverrides
-  Firewall: !ruby/object:Provider::Ansible::ResourceOverride
-    exclude: true
+  BackendService: !ruby/object:Provider::Ansible::ResourceOverride
+    aliases:
```
[the remark this answers is not in the document]
Comment (rambleraptor) [reply products/compute/ansible.yaml]: Good call! Fixed. Much more property_override driven.

File: templates/ansible/resource.erb (hunk 1 above)
Comment (rosbo) [reply templates/ansible/resource.erb]: Ping

Comment (rambleraptor) [reply templates/ansible/resource.erb]: (But actually fixed now)

File: products/dns/api.yaml (hunk 3 above)
Comment (rambleraptor) [reply products/dns/api.yaml]: Updated.

(Chef DNS did not previously have a changelog).

**Pull request merged**